### PR TITLE
Update all of plflib packages

### PR DESCRIPTION
**Pull request by Cons-Cat**
I have updated `plf_colony` (which changed its versioning scheme to be compatible with vcpkg's), and added plflib libraries which were missing for some reason. I have also added a `plf` metapackage to simply grab all of plflib at once.

### CODE REVIEW

**Comment by Cons-Cat** [comment]
The issue appears to be that Visual Studio 2019 cannot build `plf_hive`, however Visual Studio 2022 can. I'm not sure how to properly fix this. @waruqi Do you have advice?

**Comment by waruqi** [comment]
I think that you should split into separate patches, with one pr submitting one package, to make it easier to debug and merge.

**Comment by Cons-Cat** [comment]
As far as I can tell, all the errors are coming from `plf::hive`. The libraries in general are extremely compatible, even for C++03, _except_ for `plf::hive` which is pretty modern. I've removed that from the PR for now.

File: packages/p/plflibs/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    on_install(function (package)
+        package:add("deps", "plf_colony")
+        package:add("deps", "plf_indiesort")
+        package:add("deps", "plf_list")
+        package:add("deps", "plf_queue")
+        package:add("deps", "plf_rand")
+        package:add("deps", "plf_reorderase")
+        package:add("deps", "plf_stack")
+    end)
```
Comment (star-hengxing) [inline packages/p/plflibs/xmake.lua:17]: Unnecessary code, can we remove it?

Comment (waruqi) [reply packages/p/plflibs/xmake.lua:17]: sure

**Pull request merged**